File: Semestr_IV/Programowanie_Obiektowe_II/oop-ts-shop-old/src/oop/Product.ts
export class Product {
    private readonly _id: string;
    private readonly _name: string;
    private readonly _price: number;

    constructor(id: string, name: string, price: number) {
        if (!id) throw new Error("Product id cannot be empty");
        if (!name) throw new Error("Product name cannot be empty");
        if (price < 0) throw new Error("Price cannot be negative");

        this._id = id;
        this._name = name;
        this._price = price;
    }

    get id(): string {
        return this._id;
    }

    get name(): string {
        return this._name;
    }

    get price(): number {
        return this._price;
    }
}

File: Semestr_IV/Programowanie_Aplikacji_Desktopowych/Projekt_Zaliczeniowy/Warehouse/Models/Product.cs
using System.Collections.Generic;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using Warehouse.Models.Types;

namespace Warehouse.Models
{
    [BsonIgnoreExtraElements]
    public class Product
    {
        [BsonId]
        public ObjectId Id { get; set; }
        public string Barcode { get; set; } = string.Empty;
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public string CategoryId { get; set; } = string.Empty;
        public int Quantity { get; set; }
        public Price Price { get; set; } = new(); 
        public Measurand Measurand { get; set; } = new();
        public byte[]? ImageData { get; set; }
        public byte[]? LabelImageData { get; set; }
        public string ExtractedLabelText { get; set; } = string.Empty;
    }
}